src/prefect/_internal/concurrency/supervisors.py

```diff
@@ -0,0 +1,212 @@
+"""
+Implementations of supervisors for futures, which allow work to be sent back to the
+thread waiting for the result of the future.
+"""
```

Comment:
This whole module is clutch, :a: :heavy_plus_sign: 

These primitives really get to the heart of what client-side orchestration is all about.  It's like a micro-scale version of the hybrid architecture.  It's _your_ thread/event_loop, and you're asking `prefect` to advise you what to run on it.  `prefect` is a passenger navigating, not driving the car.  Love it!